src/test/java/com/tngtech/configbuilder/annotation/typetransformer/StringToPrimitiveTransformerTest.java

```diff
@@ -29,7 +30,12 @@ public class StringToPrimitiveTransformerTest {
 
     @Test
     public void testTransform() {
-        initializeFactoryAndHelperMocks();
+        when(configBuilderFactory.getInstance(ErrorMessageSetup.class)).thenReturn(errorMessageSetup);
+        when(configBuilderFactory.getInstance(GenericsAndCastingHelper.class)).thenReturn(genericsAndCastingHelper);
+        when(genericsAndCastingHelper.castTypeToClass(boolean.class)).thenReturn((Class)boolean.class);
+        when(genericsAndCastingHelper.castTypeToClass(double.class)).thenReturn((Class)double.class);
+        when(genericsAndCastingHelper.castTypeToClass(int.class)).thenReturn((Class)int.class);
```

Comment:
_Unrelated to your changes, but as I already had to read this test:_ 😉

I see that a unit test for `StringOrPrimitiveToPrimitiveTransformer` should (be called `StringOrPrimitiveToPrimitiveTransformerTest` 😜 and) ideally not include its dependencies, but for a class like `GenericsAndCastingHelper` (which feels more like a static utility than a service dependency to me), for which the mock needs to be configured to return the real results, I would just pragmatically include the actual implementation:
```java
when(configBuilderFactory.getInstance(GenericsAndCastingHelper.class))
    .thenReturn(new GenericsAndCastingHelper());
```
and save a lot of boring boilerplate code. If you like the idea: I've prepared a suggestion on the branch `upgrade_to_junit5+simplify_StringOrPrimitiveToPrimitiveTransformerTest`.  (If not: that's okay, too! 😉)

Reply:
Thanks, I'll integrate it!